src/mpi/romio/mpi-io/mpir_cst_filesys.c

```diff
@@ -13,18 +13,90 @@
 #include <stdlib.h>
 #endif
 
-/* not to be called directly (note the MPIR_ prefix), but instead from
- * MPI-level MPI_Comm_split_type implementation (e.g.
- * MPIR_Comm_split_type_impl). */
-#undef FUNCNAME
-#define FUNCNAME MPIR_Comm_split_filesystem
-#undef FCNAME
-#define FCNAME MPL_QUOTE(FUNCNAME)
+/* TODO: configure check and guard these */
```

Comment:
We've already got a check for `sys/types.h`. Portability for `dirent.h/opendir` seems to be okay across our testing environment, but it would still be nice to have an escape hatch at configure time to avoid any build issues.